### PR TITLE
fix: Fix improper policy checking

### PR DESCRIPTION
**Pull request by filipecabaco**
## What kind of change does this PR introduce?

* Presence had a bad check when tracking changes
* JWT checking had a bad check and a broken test
* Fixed some migrations to be less susceptible to errors on migrations
* Added one connection to db_pool
* Removed one connection from subcriber_pool_size

### CODE REVIEW

File: lib/realtime_web/channels/realtime_channel.ex [hunk 1]
```diff
@@ -239,6 +239,8 @@ defmodule RealtimeWeb.RealtimeChannel do
 
   @impl true
   def handle_info(:confirm_token, %{assigns: %{pg_change_params: pg_change_params}} = socket) do
+    IO.inspect(socket.policies, label: "confirm_token")
+
```
Comment (chasers) [inline lib/realtime_web/channels/realtime_channel.ex:243]:
```suggestion
```

File: lib/realtime_web/channels/realtime_channel.ex [hunk 2]
```diff
@@ -486,19 +493,17 @@ defmodule RealtimeWeb.RealtimeChannel do
          exp_diff when exp_diff > 0 <- exp - Joken.current_time(),
          {:ok, socket} <- validate_policy(socket, claims, check_policy) do
```
Comment (chasers) [inline lib/realtime_web/channels/realtime_channel.ex:494]: In the join we have a pipe of (pseudo) `|> confirm_token |> assign_policy` and it's obvious that policies get assigned to the socket this way.

I would call `assign_policy` from the `access_token` `handle_in` callback so it's obvious policies are assigned there.

Right now this `validate_policy` is maybe called, and then it maybe calls `assign_policy`.

**Review by chasers — CHANGES_REQUESTED** [review]

File: lib/realtime_web/channels/realtime_channel.ex (hunk 2 above)
Comment (filipecabaco) [reply lib/realtime_web/channels/realtime_channel.ex]: did a refactor to check the policy jwt change, could you recheck?

File: lib/realtime_web/channels/realtime_channel.ex [hunk 3]
```diff
@@ -312,25 +312,25 @@ defmodule RealtimeWeb.RealtimeChannel do
...
-            _ -> nil
-          end
-
-        assigns = %{
-          pg_sub_ref: pg_sub_ref,
-          confirm_token_ref: confirm_token_ref,
-          pg_change_params: pg_change_params
-        }
+    with {:ok, claims, confirm_token_ref, _, socket} <- confirm_token(socket),
+         {:ok, socket} <- validate_policy(socket, claims) do
```
Comment (chasers) [inline lib/realtime_web/channels/realtime_channel.ex:316]: Call `validate_policy` `maybe_assign_policy`? It's not really validating anything.

**Review by chasers — APPROVED** [review]

File: lib/realtime_web/channels/realtime_channel.ex [hunk 4]
```diff
@@ -310,24 +312,27 @@ defmodule RealtimeWeb.RealtimeChannel do
...
-        pg_sub_ref =
-          case pg_change_params do
-            [_ | _] -> postgres_subscribe()
-            _ -> nil
-          end
+      pg_sub_ref =
+        case pg_change_params do
+          [_ | _] -> postgres_subscribe()
+          _ -> nil
+        end
```
Comment (abc3) [inline lib/realtime_web/channels/realtime_channel.ex:324]:
```suggestion
      pg_sub_ref = if pg_change_params != [], do: postgres_subscribe()
```

File: lib/realtime_web/channels/realtime_channel.ex [hunk 5]
```diff
@@ -249,9 +249,11 @@ defmodule RealtimeWeb.RealtimeChannel do
            pg_change_params: pg_change_params
          })}
 
-      {:error, error} ->
-        message = "Access token has expired: " <> Helpers.to_log(error)
+      {:error, error} when is_binary(error) ->
+        shutdown_response(socket, error)
 
+      {:error, error} ->
+        message = Helpers.to_log(error)
```
Comment (abc3) [inline lib/realtime_web/channels/realtime_channel.ex:256]: just checking if really don't need "Access token has expired: "

File: lib/realtime_web/channels/realtime_channel.ex [hunk 6]
```diff
@@ -310,24 +312,27 @@ defmodule RealtimeWeb.RealtimeChannel do
...
+        pg_sub_ref: pg_sub_ref,
+        confirm_token_ref: confirm_token_ref,
+        pg_change_params: pg_change_params
+      }
 
-        {:noreply, assign(socket, assigns)}
+      {:noreply, assign(socket, assigns)}
+    else
+      {:error, error} when is_binary(error) ->
+        shutdown_response(socket, error)
```
Comment (abc3) [inline lib/realtime_web/channels/realtime_channel.ex:335]: missed "Received an invalid access token from client:"?
```suggestion
        shutdown_response(socket, "Received an invalid access token from client: " <> error)
```

File: lib/realtime_web/channels/realtime_channel.ex (hunk 5 above)
Comment (filipecabaco) [reply lib/realtime_web/channels/realtime_channel.ex:256]: In this case I'm not sure honestly 🤔

File: lib/realtime_web/channels/realtime_channel.ex (hunk 6 above)
Comment (filipecabaco) [reply lib/realtime_web/channels/realtime_channel.ex:338]: I think we might need to rethink this error 🤔 more on the side of "Your token as been invalidated" or something like that

**Review by chasers — APPROVED** [review]

File: lib/realtime_web/channels/realtime_channel.ex (hunk 5 above)
Comment (filipecabaco) [reply lib/realtime_web/channels/realtime_channel.ex]: added back

File: lib/realtime_web/channels/realtime_channel.ex (hunk 4 above)
Comment (filipecabaco) [reply lib/realtime_web/channels/realtime_channel.ex:327]: will probably tackle this in another PR

**Comment by fullstackwebdev** [comment]
this is broken code

**Pull request merged**